dissect/target/plugins/os/unix/trash.py

```diff
@@ -0,0 +1,121 @@
+from typing import Iterator
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers import configutil
+from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record import create_extended_descriptor
+from dissect.target.plugin import Plugin, alias, export
+from dissect.target.plugins.general.users import UserDetails
+from dissect.target.target import Target
+
+TrashRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+    "linux/filesystem/recyclebin",
+    [
+        ("datetime", "ts"),
+        ("path", "path"),
+        ("filesize", "filesize"),
+        ("path", "deleted_path"),
+        ("path", "source"),
+    ],
+)
+
+
+class GnomeTrashPlugin(Plugin):
+    """Linux GNOME Trash plugin.
+
+    Recovers deleted files and artifacts from ``$HOME/.local/share/Trash``.
+    Probably also works with other desktop interfaces as long as they follow the Trash specification from FreeDesktop.
+
+    Currently does not parse media trash locations such as ``/media/$Label/.Trash-1000/*``.
+
+    Resources:
+        - https://specifications.freedesktop.org/trash-spec/latest/
+        - https://github.com/GNOME/glib/blob/main/gio/glocalfile.c
+        - https://specifications.freedesktop.org/basedir-spec/latest/
+    """
+
+    PATHS = [
+        # Default $XDG_DATA_HOME/Trash
+        ".local/share/Trash",
+    ]
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.trashes = list(self.garbage_collector())
+
+    def garbage_collector(self) -> Iterator[tuple[UserDetails, TargetPath]]:
```

Comment:
```suggestion
    def _garbage_collector(self) -> Iterator[tuple[UserDetails, TargetPath]]:
```
Although I love the name, this is somewhat more consistent with the rest of these type of functions.